queries/latex/highlights.scm

```diff
@@ -313,6 +298,15 @@
       (_) @markup.link.url))
   (#any-of? @_name "\\url" "\\href"))
 
+[
+  (line_comment)
+  (block_comment)
+  (comment_environment)
+] @comment @spell
+
+((line_comment) @keyword.directive
+  (#lua-match? @keyword.directive "^%% !TeX"))
```

Comment:
Can this line directive happen everywhere, or only the first few lines of the tex file

If it's the latter then I suggest limiting it to first few line at top-level only

Reply:
I am not sure, I searched around but couldn't find whether it can only be used at the top (though I think it usually is only used at the top). Maybe someone of more expertise can help here

Reply:
This, so-called the TeX "magic comments", should come in the first few lines *by convention*, although it's not strictly mandatory from a technical perspective.

https://tex.stackexchange.com/questions/78101/when-and-why-should-i-use-tex-ts-program-and-tex-encoding/84687#84687
https://www.texdev.net/2011/03/24/texworks-magic-comments/

Reply:
Another special comment is for pre-compiled headers. (This is a LaTeX feature, and not tied to TeX works or other conventions.) A document that starts with

    %& filename

signals the document is using precomplied headers, and tex looks for filename.fmt

Would it be possible to add these special comments here too? Or do they go into the grammar.js upstream?

Thanks again for fixing this so quickly!

Reply:
PS: Comments can come before the `%&` comment; but I don't think any regular commands can. I didn't find a reference for this (but tested it).